app/scripts/collections/collection.directive.js

```diff
@@ -45,15 +45,15 @@ bulletApp.directive('collection', function($log){
             scope.removeBullet = function(bullet) {
                 scope.collection.removeBullet(bullet)
                 .then(function(){
-                    scope.bullets = scope.bullets.filter(b => b.id !== bullet.id);
+                    scope.collection.bullets = scope.collection.bullets.filter(b => b.id !== bullet.id);
                 })
                 .catch($log.err);
             };
+
             scope.addBullet = function(bullet) {
                 if (bullet.content.length > 0) {
                   scope.collection.addBullet(bullet)
                   .then(function(){
-                      scope.bullets.push(bullet);
                       scope.newBullet = new Bullet.Task()
                       scope.$evalAsync()
```

Comment:
Consider moving this digesting logic into the factory methods if possible, I'm thinking as a `$rootScope.$evalAsync`. This is just so you don't have to keep track of if/when to digest all over the place.